### PR TITLE
Mount only selected spaces

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

---

**Related issue :**

### CODE REVIEW

**Comment by enolfc** [comment]
This is deployed at training.notebooks.egi.eu. It does show this in the log when spawning
```
[E 2021-08-23 11:47:45.090 JupyterHub web:1789] Uncaught exception GET /hub/api/users/529a87e5ce04cd5ddd7161734d02df0e2199a11452430803e714cb1309cc3907@egi.eu/server/progress (10.36.0.0)
    HTTPServerRequest(protocol='https', host='training.notebooks.egi.eu', method='GET', uri='/hub/api/users/529a87e5ce04cd5ddd7161734d02df0e2199a11452430803e714cb1309cc3907@egi.eu/server/progress', version='HTTP/1.1', remote_ip='10.36.0.0')
    Traceback (most recent call last):
      File "/usr/local/lib/python3.8/dist-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/apihandlers/users.py", line 661, in get
        async for event in events:
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/utils.py", line 545, in iterate_until
        yield item_future.result()
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/spawner.py", line 1068, in _generate_progress
        async for event in progress:
      File "/usr/local/lib/python3.8/dist-packages/kubespawner/spawner.py", line 2108, in progress
        start_future = self._start_future
    AttributeError: 'DataHubSpawner' object has no attribute '_start_future'

[E 2021-08-23 11:47:45.094 JupyterHub web:1197] Cannot send error response after headers written
```

But seems not related with the added code as commenting it out also causes the problem.

File: egi_notebooks_hub/egiauthenticator.py [hunk 1]
```diff
@@ -193,28 +174,30 @@ class DataHubAuthenticator(EGICheckinAuthenticator):
...
-        http_client = AsyncHTTPClient()
+    async def create_onedata_token(self, checkin_token, token_name, caveats):
         onedata_token = None
         onedata_user = None
-        # We now go to the datahub to get a token
-        checkin_token = user_data["auth_state"]["access_token"]
+        http_client = AsyncHTTPClient()
         headers = {
             "content-type": "application/json",
             "x-auth-token": "egi:%s" % checkin_token,
```
Comment (andrea-manzi) [inline egi_notebooks_hub/egiauthenticator.py:197]: should this be configurable as welll? imagine deployments of Onedata not configured with checkin ( e.g.  Panosc)

Comment (enolfc) [reply egi_notebooks_hub/egiauthenticator.py:197]: ok, rethinking this

Comment (enolfc) [reply egi_notebooks_hub/egiauthenticator.py]: take a look now. I also moved things to a separate onedata.py file

File: egi_notebooks_hub/onedata.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+from egi_notebooks_hub.egiauthenticator import EGICheckinAuthenticator
+from egi_notebooks_hub.egispawner import EGISpawner
+from tornado.httpclient import AsyncHTTPClient, HTTPError, HTTPRequest
+from traitlets import Bool, Dict, List, Unicode
+
+
+class OnedataAuthenticator(EGICheckinAuthenticator):
+    """
+    Datahub authenticator for JupyterHub
```
Comment (andrea-manzi) [inline egi_notebooks_hub/onedata.py:15]:
```suggestion
    Onedata authenticator for JupyterHub
```

File: egi_notebooks_hub/onedata.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
+"""
+DataHub extras for the Oauthenticator
```
Comment (andrea-manzi) [inline egi_notebooks_hub/onedata.py:2]:
```suggestion
Onedata extras for the Oauthenticator
```

**Review by andrea-manzi — APPROVED** [review]
changes already deployed on the training instance, everything looks good

**Pull request merged**